geonode/layers/views.py

```diff
@@ -310,7 +310,10 @@ def layer_upload(request, template='upload/layer_upload.html'):
                     except TypeError:
                         upload_session.traceback = traceback.format_tb(tb)
                     upload_session.context = log_snippet(CONTEXT_LOG_FILE)
-                    upload_session.save()
+                    try:
+                        upload_session.save()
+                    except ValueError:
```

Comment:
catches a Null encoding error, it seems to bring the stack trace to the next error. The error code from terminal will change to 400 after this modification.